week07/todolist/src/components/Clock.jsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+
+const Clock = () => {
+    const [formattedDate, setFormattedDate] = useState("0000.00.00.");
+    const [formattedTime, setFormattedTime] = useState("00:00:00");
+
+    useEffect(() => {
+        const updateClock = () => {
+            const date = new Date();
+
+            const padHours = String(date.getHours()).padStart(2, "0");
+            const padMinutes = String(date.getMinutes()).padStart(2, "0");
+            const padSeconds = String(date.getSeconds()).padStart(2, "0");
+
+            setFormattedDate(
+                `${date.getFullYear()}. ${
+                    date.getMonth() + 1
+                }. ${date.getDate()}.`
+            );
+            setFormattedTime(`${padHours}:${padMinutes}:${padSeconds}`);
+        };
+
```

Comment:
P2: Date와 Time을 하나의 객체로 관리하면 중복되는 코드를 줄이고, 보다 효율적인 상태 관리를 할 수 있을 것 같아요!

```suggestion
 const [dateTime, setDateTime] = useState({
    formattedDate: "0000.00.00.",
    formattedTime: "00:00:00",
  });
  useEffect(() => {
    const updateClock = () => {
      const date = new Date();
      const formattedDate = `${date.getFullYear()}.${
        date.getMonth() + 1
      }.${date.getDate()}.`;
      const formattedTime = [
        date.getHours(),
        date.getMinutes(),
        date.getSeconds(),
      ]
        .map((num) => String(num).padStart(2, "0"))
        .join(":");

      setDateTime({ formattedDate, formattedTime });
    };

    const timer = setInterval(updateClock, 1000);
    return () => clearInterval(timer);
  }, []);

  return (
    <div className='clock'>
      <div className='day'>{dateTime.formattedDate}</div>
      <div className='time'>{dateTime.formattedTime}</div>
    </div>
  );
```

이런식으로 time과 day를 dateTime이라는 하나의 객체로 관리할 수도 있습니다.